src/lib/utils.py

```diff
@@ -68,6 +68,15 @@ def _extract_timestamp(info):
     raise CheckerRemoteError(f"Cannot parse date/time: {date_str}")
 
 
+def _check_newline(fp):
+    fp.seek(0, 2)
+    fp.seek(fp.tell()-1,0)
```

Comment:
Please use named constants instead of numeric values, e.g. `os.SEEK_END` instead of `2`.
Also I guess this can be done in one action with negative offset instead of two, i.e. `fp.seek(-1, os.SEEK_END)`?